inject-java-test/build.gradle

```diff
@@ -29,6 +29,14 @@ dependencies {
     testImplementation libs.smallrye
 }
 
+tasks.withType(JavaCompile).configureEach {
+    options.compilerArgs += [
+            '--add-exports', 'jdk.compiler/com.sun.tools.javac.main=ALL-UNNAMED'
```

Comment:
:x: you don't need this. The reason it happens is because of this code: https://github.com/micronaut-projects/micronaut-core/blob/4.0.x/inject-java-test/src/main/java/io/micronaut/annotation/processing/test/JavaParser.java#L340-L347

But as the comment states this is only required for Java <15 and that we're building with 17, this whole block can be removed.

Reply:
But then `inject-java-test:compileJava` fails with

https://ge.micronaut.io/s/k3o44k23ckrfq/console-log?task=:inject-java-test:compileJava

Reply:
Ahhhhhh...  Sorry about that! ☺️

Done in https://github.com/micronaut-projects/micronaut-core/pull/8154/commits/9fb60e2adccf024338487a34259456b88e1f1e68